msal/src/test/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationKotlinTest.kt

```diff
@@ -2293,3 +2408,48 @@ class NativeAuthPublicClientApplicationKotlinTest : PublicClientApplicationAbstr
         assertTrue((result as SignUpError).isInvalidUsername())
     }
 }
+
+class LoggerTestCallback(private val infoPIIToCheck: List<String>, private val allowPIIFalseToCheck: List<String>, private val allowPII: Boolean, private val latch: CountDownLatch?) : ILoggerCallback {
+    var failCalled = false
+    var count = 0  // Used to know how many logs need to be checked
+    override fun log(
+        tag: String?,
+        logLevel: Logger.LogLevel?,
+        message: String?,
+        containsPII: Boolean
+    ) {
+        if (allowPII) {
+            allowPIIFalseToCheck.forEach { regex ->
```

Comment:
This part might be not necessary because LoggerTest takes the responsibility of validating containsPII.

Reply:
Not exactly. LoggerTest validates whether, if a native auth object states that it contains PII (through `ILoggable`), the callback's `containsPII` is set to `true` (and vice versa). But there are more ways through which `containsPII` can be set. Moreover: these tests validate whether `containsPII` is correctly set according to actual fields (like username, attributes, etc.), and they're run on a wide range of scenarios. This is more of an E2E test, whereas `LoggerTest` is a unit test that tests the `ILoggable` interface specifically.

Reply:
Add verifyLogCouldContain function to validate whether containsPII is correctly set according to actual fields.